tests/test_esm_parser/test_provenance.py

```diff
@@ -2,143 +2,129 @@
     Unit tests for the new provenance feature
 """
 import os
-import esm_parser.provenance as provenance
 import pathlib
+import esm_parser.provenance as provenance
 import esm_parser
 from esm_parser import yaml_to_dict
 
-config_dict = {
-    "echam": {
-        "type": "atmosphere",
-        "files": {
-            "greenhouse": {"kind": "input", "path_in_computer": "/my/path/in/computer"}
-        },
-    }
-}
+#config_dict = {
+#    "echam": {
+#        "type": "atmosphere",
+#        "files": {
+#            "greenhouse": {"kind": "input", "path_in_computer": "/my/path/in/computer"}
+#        },
+#    }
+#}
+#
+#my_provenance = {
+#    "echam": {
+#        "type": {"line": 2, "col": 11, "yaml_file": "myrunscript.yaml", "category": "runscript"},
+#        },
+#}
+#config = provenance.DictWithProvenance(config_dict, my_provenance)
 
-my_provenance = {
-    "from_file": None,
-    "type": None,
+#os.environ['GITHUB_WORKSPACE'] = "/work/ab0995/a270089/esm_tools"
+config = yaml_to_dict.yaml_file_to_dict(str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example2.yaml").resolve()))
+
+check_provenance = {'echam':
+    {'type':
+        {'line': 2, 'col': 11, 'yaml_file': str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example2.yaml").resolve()), 'category': 'runscript'},
+        'files': {
+            'greenhouse': {
+                'kind': {'line': 5, 'col': 19, 'yaml_file': str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example2.yaml").resolve()), 'category': 'runscript'},
+                'path_in_computer': {'line': 6, 'col': 31, 'yaml_file': str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example2.yaml").resolve()), 'category': 'runscript'}
+            }
+        }
+    },
+    'debug_info': {'loaded_from_file': None}
 }
-config = provenance.DictWithProvenance(config_dict, my_provenance)
+# Test 1:  Checks for a correct provenance entries from example yaml file.
+def test_get_provenance_from_yaml_to_dict():
+    assert config.get_provenance() == check_provenance
```

Comment:
Turn the comment into the function's docstring
```suggestion
def test_get_provenance_from_yaml_to_dict():
    """
    Checks for a correct provenance entries from example yaml file
    """
    assert config.get_provenance() == check_provenance
```

Reply:
Done